### PR TITLE
[CodeQuality] fix some defects to improve code quality

### PR DESCRIPTION
**Pull request by helloahn**
There are a few code changes to improve the code quality.
- check the variables before using them.
- return error code.
- free the allocated memory when it finished with ASSERT.

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

Signed-off-by: Hyoung Joo Ahn <hello.ahn@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_decoder/tensordec-imagelabel.c [hunk 1]
```diff
@@ -202,6 +202,11 @@ il_decode (void **pdata, const GstTensorsConfig * config,
   str = data->labels.labels[max_index];
   size = strlen (str);
 
+  if (!str || size == 0) {
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_decoder/tensordec-imagelabel.c:205]: recommend to change line 203 ~ 205 (IMO, the tool may check strlen (nullptr) case if this condition is added)
```
if (!str || (size = strlen (str)) == 0) {
```

File: tests/nnstreamer_plugins/unittest_plugins.cc [hunk 2]
```diff
@@ -3650,8 +3653,11 @@ TEST (testTensorTransform, orcPerformance)
   diff_loop = stop_ts - start_ts;
   _print_log ("combined loop: %" G_GINT64_FORMAT, diff_loop);
 
+error:
   g_free (data_u8);
   g_free (data_float);
+  
+  ASSERT_TRUE(ret);
```
Comment (gichan-jang) [inline tests/nnstreamer_plugins/unittest_plugins.cc:3660]:
```suggestion
  ASSERT_TRUE (ret);
```

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 3]
```diff
@@ -431,6 +431,10 @@ gst_tensor_filter_get_layout_string (const GstTensorFilterProperties * prop,
 static gchar *
 strcpy2 (gchar * dest, const gchar * src)
 {
+  if (!src) {
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c:434]: dest also should be checked.
```
if (!desc || !src)
```

File: tests/nnstreamer_plugins/unittest_plugins.cc (hunk 2 above)
Comment (helloahn) [reply tests/nnstreamer_plugins/unittest_plugins.cc:3660]: nice catch :+1:

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c (hunk 3 above)
Comment (helloahn) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c:434]: got it

File: ext/nnstreamer/tensor_decoder/tensordec-imagelabel.c (hunk 1 above)
Comment (helloahn) [reply ext/nnstreamer/tensor_decoder/tensordec-imagelabel.c]: looks better

**Review by gichan-jang — APPROVED** [review]

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**